src/gui/src/PatternPropertiesDialog.cpp

```diff
@@ -125,45 +128,11 @@ void PatternPropertiesDialog::on_okBtn_clicked()
 }
 
 void PatternPropertiesDialog::defaultNameCheck( QString pattName, bool savepattern)
-{
-	if ( savepattern && !nameCheck(pattName) )
-	{
-		PatternList *pPatternList = Hydrogen::get_instance()->getSong()->getPatternList();
-		QString newPattName = pPatternList->find_unused_pattern_name(pattName);
-		patternNameTxt->setText( tr( "%1").arg(newPattName) );
-	}
-	else
-	{
-		patternNameTxt->setText( tr( "%1").arg(pattName) );
-	}
-}
-
-
-bool PatternPropertiesDialog::nameCheck( QString pattName )
 {
 	PatternList *pPatternList = Hydrogen::get_instance()->getSong()->getPatternList();
-	
-	return pPatternList->check_name(pattName);
-}
-
-
-void PatternPropertiesDialog::on_categoryComboBox_editTextChanged()
-{
-	if ( categoryComboBox->currentText() == pattern->get_category() ) {
-		okBtn->setEnabled( false );
-	}
-	else {
-		okBtn->setEnabled(true);
+	if ( savepattern && !pPatternList->check_name(pattName, pattern) ) {
+		pattName = pPatternList->find_unused_pattern_name(pattName, pattern);
 	}
-}
-
 
-void PatternPropertiesDialog::on_patternNameTxt_textChanged()
-{
-	if ( nameCheck( patternNameTxt->text() ) ) {
-		okBtn->setEnabled( true );
-	}
-	else {
-		okBtn->setEnabled( false );
-	}
+	patternNameTxt->setText( tr("%1").arg(pattName) );
```

Comment:
Could you remove the `tr()` around "%1"? This marks the string "%1" translatable and it will show up in `.ts` file intended for the translators. No idea why this had been there in the first place

Reply:
Actually, is the call to `arg()` needed here at all, or could the whole line just be `patternNameTxt->setText(pattName)`?

Reply:
Yes, your right.